lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -976,6 +977,18 @@ SubExprInfo ExprEmitter::visitComb(ConstantOp op) {
   return {Unary, signPreference == RequireSigned ? IsSigned : IsUnsigned};
 }
 
+SubExprInfo ExprEmitter::visitComb(ArraySliceOp op) {
+  auto arrayPrec = emitSubExpr(op.input(), Symbol);
+
+  unsigned dstWidth = op.getType().cast<ArrayType>().getSize();
+  os << '[';
+  emitSubExpr(op.lowIndex(), LowestPrecedence);
+  os << '+' << dstWidth - 1 << ':';
+  emitSubExpr(op.lowIndex(), LowestPrecedence);
+  os << ']';
```

Comment:
Should this handle the "extract exactly one element" case without the colon?

Reply:
I would find that confusing if I was reading the output, so I don't think so.

Reply:
Agreed, I forgot that this was explicitly a slice; we should have an "index" operation that returns a single element and strength reduce one into the other.